core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -973,8 +1057,11 @@ private void initAfterRebalance() {
         /**
          * Advances next to higher entry.
          * Return previous index
+         *
+         * @return The first long is the key's reference, the integer is the value's version and the second long is
+         * the value's reference. If {@code needsValue == false}, then the value of the map entry is {@code null}.
          */
-        Map.Entry<ByteBuffer, Slice> advance(boolean needsKey, boolean needsValue) {
+        Map.Entry<Long, Map.Entry<Integer, Long>> advance(boolean needsValue) {
```

Comment:
Map Entry on top of the other Map Entry? Here it looks to me a good place to add a class... Or can we use the existing Result?

Reply:
The inner map entry is the version of the value and its reference.
I could create a class which is a pair of int and long if needs be.